go/database/mpt/proof_test.go

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package mpt
+
+import (
+	"errors"
+	"fmt"
+	"github.com/Fantom-foundation/Carmen/go/common"
+	"github.com/Fantom-foundation/Carmen/go/database/mpt/shared"
+	"go.uber.org/mock/gomock"
+	"golang.org/x/exp/maps"
+	"strings"
+	"testing"
+)
+
+func TestCreateWitnessProof_CanGenerateProof(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	address := common.Address{0xAB, 0xCD, 0xEF}
+	key1 := common.Key{0x12, 0x34, 0x56, 0x78}
+	key2 := common.Key{0x23, 0x45, 0x67, 0x78}
+
+	ctxt := newNodeContext(t, ctrl)
+	root, node := ctxt.Build(&Branch{
+		children: Children{
+			0xA: &Branch{
+				children: Children{
+					0xB: &Extension{
+						path: []Nibble{0xC, 0xD, 0xE},
+						next: &Account{address: address, info: AccountInfo{common.Nonce{1}, common.Balance{1}, common.Hash{0xAA}},
+							storage: &Branch{
+								children: Children{
+									0x1: &Extension{path: []Nibble{0x2, 0x3, 0x4}, next: &Value{key: key1, value: common.Value{0x12}}},
+									0x2: &Extension{path: []Nibble{0x3, 0x4, 0x5}, next: &Value{key: key2, value: common.Value{0x34}}},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	})
+
+	proof, err := CreateWitnessProof(ctxt, &root, address, key1, key2)
+	if err != nil {
+		t.Fatalf("failed to create proof: %v", err)
+	}
+
```

Comment:
nit: check `proof.IsValid()` and `proof.GetBalance(...)` or `proof.GetStorage(...)` ... maybe here or at another place.

Reply:
ok, but they are not implemented yet, perhaps I may postpone this PR when the methods are ready.